<?xml version="1.0"?>
<launch>

  <arg name="node_name" default="gravity_compensation"/>
  <arg name="calib_data_file" default="$(find gravity_compensation)/config/ft_calib_data.yaml"/>
  <arg name="ft_raw" default="robotiq_ft_wrench"/>
  <arg name="imu" default="imu/data"/>
  <arg name="loop_rate" default="1000.0"/>
  <arg name="ns" default="ft_sensor"/>
  <arg name="gripper_com_child_frame_id" default="robotiq_ft_frame_id"/>
  <arg name="use_tf" default="true"/>

  <node pkg="tf2_ros" type="static_transform_publisher" name="ft_frame_pub" args="0 0 0 0 0 0 RARM_JOINT5_Link robotiq_ft_frame_id" />

  <group if="$(arg use_tf)">
    <node pkg="gravity_compensation" type="tf_to_imu.py" name="tf_to_imu" output="screen">
      <param name="base_frame" value="WAIST"/>
      <param name="sensor_frame" value="robotiq_ft_frame_id"/>
      <remap from="/imu/data" to="/imu/data"/>
    </node>
  </group>
  
  <node name="$(arg node_name)" pkg="jsk_nextage_bringup" type="gravity_compensation_node" cwd="node" respawn="false" output="screen">
    <param name="wait_for_tf" value="true"/>
    <rosparam command="load" file="$(arg calib_data_file)"/>
    <remap from="$(arg node_name)/ft_raw" to="$(arg ft_raw)" />
    <remap from="$(arg node_name)/imu" to="$(arg imu)" />
    <param name="loop_rate" value="$(arg loop_rate)" />
    <param name="ns" value="$(arg ns)"/>
    <param name="gripper_com_child_frame_id" value="$(arg gripper_com_child_frame_id)"/>
  </node>

</launch>
